### PR TITLE
Custom sites update

### PR DESCRIPTION
**Pull request by iMattPro**
This got it working where all you need to do is add stand alone XML files for new sites.

It's working, but the coding is still kind of bad, and probably wrong. But maybe Joshy can help us when this is merged in your PR.

Maybe there's a way these can be appended to the default sites, so the coding inside the ACP and the event listener won't have to change much at all.

### CODE REVIEW

**Comment by rxu** [comment]
I was about getting rid of the ```customsitecollection``` class at all.

**Comment by iMattPro** [comment]
Perhaps but then we need the root to make the path everywhere we'd use it, so this is kind of nice to have a single place for that.

**Comment by iMattPro** [comment]
Like if you noticed I just gave up totally in the event listener and used a global to get the phpbb root path...just to see if this would even work.

**Comment by rxu** [comment]
I wonder if just ```collection\xml``` would work (without the ```$phpbb_root_path``` at all).

**Comment by iMattPro** [comment]
No I tried that already

File: acp/main_module.php [hunk 1]
```diff
@@ -145,21 +143,16 @@ protected function get_sites()
...
 		{
-			$disabled = isset($configurator->BBCodes[$siteId]);
-			$sites[] = [
-				'id'		=> $siteId,
-				'name'		=> $siteConfig['name'],
-				'title'		=> $this->language->lang($disabled ? 'ACP_MEDIA_SITE_DISABLED' : 'ACP_MEDIA_SITE_TITLE', $siteId),
-				'enabled'	=> in_array($siteId, $this->get_enabled_sites()),
-				'disabled'	=> $disabled,
-			];
+			$configurator->MediaEmbed->defaultSites->add($siteId, $siteConfig);
```
Comment (rxu) [inline acp/main_module.php:151]: it looks like if do it in the listener then```$configurator->MediaEmbed->defaultSites``` will be already expanded at this point so we won't need to edit the ACP module, doesn't it?

EDIT: tested and that worked like a charm.

**Pull request merged**